sql-migration-assistant/src/sql_migration_assistant/config.py

```diff
@@ -1,16 +1,164 @@
+    }
+
+    def __init__(self, profile=None):
+        self.from_environ()
+        self.profile = (
+            os.environ.get("DATABRICKS_PROFILE") if profile is None else profile
+        )
+        self.w = get_workspace_client(self.profile)
+        self.catalog = self.config.get("CATALOG")
+        self.schema = f"{self.catalog}.{self.config.get('SCHEMA')}"
+        self.config_table = f"{self.schema}.{self.config.get('CONFIG_TABLE_NAME')}"
+        self.validate_first_setup()
+        self.con = get_db_connection(self.profile, self.warehouse.id)
+        self.from_sql()
+
+    def get_workspace_path(self):
+        return self.config.get(
+            "WORKSPACE_PATH",
+            f"/Workspace/Users/{self.w.config.username}/sql_migration_assistant_files",
+        )
+
+    def from_sql(self):
+        try:
+            self.w.tables.get(self.config_table)
+            exists = True
+        except NotFound:
+            exists = False
+
+        if not exists:
+            logger.warning(
+                f"No Config table found at {self.config_table}. Creating new one"
+            )
+            cursor = self.con.cursor()
+            cursor.execute(
+                f"Create table {self.config_table} (key STRING, value STRING);"
+            )
+            insert_query = f"""Insert into {self.config_table} values {",".join([f"('{key}', '{value}')" for key, value in self.config.items()])};"""
+            cursor.execute(insert_query)
+            cursor.close()
+        else:
+            config = pd.read_sql(
+                f"Select key, value from {self.config_table}", con=self.con
+            )
+            for _, row in config.iterrows():
+                self.config[row["key"]] = row["value"]
+
+    def from_environ(self):
+        for key in ["CATALOG", "SCHEMA", "SQL_WAREHOUSE_NAME"]:
+            self.config[key] = os.environ.get(key)
+
+    def from_yaml(self):
+        with open(yaml_path, "r") as f:
+            content = yaml.safe_load(f)
+        self.config = {**self.config, **content}
+
+    def set_config(self, key, value):
+        logger.info(f"Setting Config {key} to {value}")
+        self.set_configs({key: value})
+
+    def set_configs(self, configs: dict):
+        logger.info(f"Setting Configs {configs}")
+        cursor = self.con.cursor()
+        insert_query = f"""Insert into {self.config_table} REPLACE WHERE key in ('{"','".join(configs.keys())}') VALUES {",".join([f"('{key}', '{value}')" for key, value in configs.items()])};"""
+        cursor.execute(insert_query)
+        cursor.close()
+        self.config = {**self.config, **configs}
+
+    def get(self, key, default=None):
+        return self.config.get(key, default)
+
+    def validate_key_exists(self, key) -> list[str]:
+        if key not in self.config:
+            return [
+                f"{key} not found in config.yml, please configure it before deployment"
+            ]
+        return []
+
+    def validate_first_setup(self):
```

Comment:
initial set up should be able to create default resources - we dont want people to try the installation, get stuck, then have to go into databricks to create something, then repeat. better to automate it